server/embed/config.go

```diff
@@ -128,6 +128,12 @@ var (
 
 	// indirection for testing
 	getCluster = srv.GetCluster
+
+	// in 3.6, we are migration all the --experimental flags to feature gate and flags without the prefix.
+	// This is the mapping from the non boolean `experimental-` to the new flags.
+	ExperimentalFlagMigrationMap = map[string]string{
+		"experimental-compact-hash-check-time": "compact-hash-check-time",
```

Comment:
Is this first flag that we are migrating? Have we used different mechanism for other flags?

Reply:
It seems that yes. It's the first non-boolean experimental flag, which needs to be migrated along with the corresponding boolean flag being migrated to server feature gate.

Since it's only used in server/embed, suggest to change it from `ExperimentalFlagMigrationMap` to `experimentalNonBoolFlagMigrationMap`, also add a TODO item "`TODO: remove it in 3.7`".

Reply:
It is also used in server/etcdmain. changing to `ExperimentalNonBoolFlagMigrationMap`

Reply:
done